src/python/friendzone/nwx2qcengine/__init__.py

```diff
@@ -11,3 +11,116 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
+from ..friends import is_friend_enabled
+import pluginplay as pp
+from simde import TotalEnergy, EnergyNuclearGradientStdVectorD
+from .call_qcengine import call_qcengine
+
+
+class QCEngineEnergy(pp.ModuleBase):
+
+    def __init__(self):
+        pp.ModuleBase.__init__(self)
+        self.satisfies_property_type(TotalEnergy())
+        self.description('Driver module for calling friends through QCEngine')
+
+        ddesc = 'Implementation detail. DO NOT MANUALLY CHANGE!'
+        self.add_input('_driver').set_description(ddesc).set_default('energy')
+        self.add_input('program').set_description('Friend to call')
+        self.add_input('method').set_description('Level of theory')
+        self.add_input('basis set').set_description('Name of AO basis set')
+
+    def run_(self, inputs, submods):
```

Comment:
A lot of this pertains to the gradient case for understandable reasons, but it's a bit weird for it to be in the module called `QCEngineEnergy`. Maybe the contents of this can be moved into an implementation function that the two classes call from their respective `run_` functions? Something like:
```python
def run_impl(driver, inputs, submods):
    # current contents of run_ go here

class QCEngineEnergy(pp.ModuleBase):
    def __init__(self):
        pp.ModuleBase.__init__(self)
        self.satisfies_property_type(TotalEnergy())
        # yada yada

    def run_(self, inputs, submods):
        results = run_impl("energy", inputs, submods)
        return results

class QCEngineGradient(pp.ModuleBase):
    def __init__(self):
        pp.ModuleBase.__init__(self)
        self.satisfies_property_type(EnergyNuclearGradientStdVectorD())
        # yada yada

    def run_(self, inputs, submods):
        results = run_impl("gradient", inputs, submods)
        return results
```
This would also eliminate the awkward `"_driver"` input.

Reply:
This is a great idea! Added in 5772741.